### ISSUE TITLE
Write generator in typescript

### ISSUE THREAD

**Issue by emileindik** [yeoman/yeoman#1734]
Is it possible to use yeoman with typescript?
If so, are there any sample generators written in typescript that you could point me to?

**Comment by mshima** [comment]
I know it’s possible because of https://github.com/yeoman/environment/issues/164.
I don’t know any public generator or example.

**Comment by StfBauer** [comment]
@emileindik The Microsoft Teams generator uses typescript you see how this project is set up there https://github.com/pnp/generator-teams

**Comment by jvanmelckebeke** [comment]
Sorry my answer is very late, I discovered this issue just today.

For clarification for future issue readers: **yes** it is possible to create a generator in typescript, however keep the following things in mind:

1. your target **JS flavour** is `esnext`
1. your target **outDir** is `lib/` ( **not** `dist/`, apparently this directory does not get scanned by the yeoman generator discovery tool)

Hope this helps for future developers who want to create a generator in typescript

**Comment by gliviu** [comment]
For anyone looking for a quick start here is a template you could use to create new ts generators [generator-typescript-template](https://github.com/gliviu/generator-typescript-template)

**Comment by juliankrieger** [comment]
@jvanmelckebeke The way I solved it is to build to `dist`, but write a script that copies the `package.json` into `dist` as well as only keeping dependencies, not dev dependencies

**Comment by chanind** [comment]
If you want to run a generator locally in a project, you can just use `ts-node` and run: `ts-node node_modules/.bin/yo ./path/to/generator`, and then your generator can be fully in typescript.

**Comment by RobertLowe** [comment]
Yeah, I want this. All said, it's really annoying to have to manage a `build` script and transpiled files, I wish this was seamless.

**Comment by Carlosf96** [comment]
> Yeah, I want this. All said, it's really annoying to have to manage a `build` script and transpiled files, I wish this was seamless.

I too wish it was seamless UwU.

**Comment by JoshuaKGoldberg** [comment]
Transferring over to the website repo as this is a docs request. Cheers! 🎩
